### DISCUSSION TITLE
[10] Dinder - RMMM

### DISCUSSION THREAD

**Discussion by Manuel-F-04** [dhbw-ka-tinf22b5-dinder/Dinder#15] [Projektblog]
# RMMM
![RMMM-Tabelle](https://github.com/dhbw-ka-tinf22b5-dinder/Dinder-SRS/blob/main/RMMM.md)

## Biggest technological risk
Our biggest risk is poor code quality (ID 2, RMMM). Thats why we use peer reviews and discuss our changes. In addition we will use test pipelines.
The aim is to achieve the highest possible test coverage in order to eliminate bugs as far as possible.
For the test execution, pipelines are to be written in github actions, which are triggered by pull requests.

## Last changes
Since Lukas Schulz is no longer part of the group, we had to redistribute his tasks. This means that we had to learn the front-end technologies that Lukas had been working on. As a result, we have not yet been able to implement any major changes to the actual program.

**Comment by ghost** [comment]
Gute detailierte Ausarbeitung eurer Risiken. Ebenfalls ist eure Erkenntnis, dass euer größtes Risiko schlechte Code Quailität ist, gelungen. Erfreulich war, dass ihr in diesem Kontext angesprochen habt, dass ihr Code Reviews und automatisierte Testausführung verwendet, um dies zu vermeiden. Eine gute Unterstützung um die Code Qualität zu gewährleisten sind statische Code Analysetools wie [Sonar](https://www.sonarsource.com/plans-and-pricing/developer/?utm_medium=paid&utm_source=bing&utm_campaign=SQ-EMEA-Brand&utm_content=SonarQube&utm_term=sonarqube&s_category=Paid&s_source=Paid%20Search&s_origin=bing&msclkid=a9d0989179f918881143fe8d69d27da9).

**Comment by NeeroxZ** [comment]
Eure proaktive Herangehensweise, technologische Risiken anzugehen, ist löööblich. Die Nutzung von Peer-Reviews und Test-Pipelines zur Verbesserung der Code-Qualität ist ein intelligenter Ansatz. Eure Priorisierung einer hohen Testabdeckung wird helfen, Bugs frühzeitig zu erkennen.

Die Umverteilung von Aufgaben im Team ist verständlich, und es ist gut, dass ihr offen über diese Änderungen kommuniziert. Ich bin zuversichtlich, dass euer Team diese Herausforderungen erfolgreich bewältigen wird.

-IncidArch
